src/main/webapp/app/exercises/shared/result/result-detail.component.ts

```diff
@@ -336,11 +268,84 @@ export class ResultDetailComponent implements OnInit {
         }
     }
 
+    /**
+     * Creates a feedback item for feedback received for a programming exercise.
+     * @param feedback The feedback from which the feedback item should be created.
+     * @private
+     */
+    private createProgrammingExerciseFeedbackItem(feedback: Feedback): FeedbackItem {
+        const previewText = ResultDetailComponent.computeFeedbackPreviewText(feedback.detailText);
+
+        if (Feedback.isSubmissionPolicyFeedback(feedback)) {
+            return this.createProgrammingExerciseSubmissionPolicyFeedbackItem(feedback, previewText);
+        } else if (Feedback.isStaticCodeAnalysisFeedback(feedback)) {
+            return this.createProgrammingExerciseScaFeedbackItem(feedback);
+        } else if (feedback.type === FeedbackType.AUTOMATIC) {
+            return this.createProgrammingExerciseAutomaticFeedbackItem(feedback, previewText);
+        } else if ((feedback.type === FeedbackType.MANUAL || feedback.type === FeedbackType.MANUAL_UNREFERENCED) && feedback.gradingInstruction) {
+            return this.createProgrammingExerciseGradingInstructionFeedbackItem(feedback, previewText);
+        } else {
+            return {
```

Comment:
Maybe use another method for this as well? Just to have every case as a one-liner.